src/main/java/pl/most/typer/service/footballservice/matches/MatchesServiceDefault.java

```diff
@@ -90,32 +83,72 @@ public HttpStatus getMatchInfoFromExternalApi(Integer competitionId) {
     }
 
     @Override
-    public void saveAll(List<Match> matches) {
+    public void saveAll(List<Match> matches) throws BadResourceException, ResourceAlreadyExistsException {
         for (Match match : matches) {
-            Optional<Match> byApiId = matchesRepository.findByApiId(match.getApiId());
-            byApiId.orElseGet(() -> matchesRepository.save(match));
+            if (!StringUtils.isEmpty(match)) {
```

Comment:
to sprawdza czy obiekt String jest pusty. A w tym przypadku ty probujesz srpawdzic obiekt Match, wiec on prawdopobnie robi na nim toString, wiec raczej pusty nigdy nie bedzie (chyba ze null??). Wiec chyba sprawdzenie tego warunku jest zbedne.